Use pluralize for catalog kind header component (#14698)
backstage/backstage: .changeset/thin-donuts-join.md

---
'@backstage/plugin-catalog': patch
---

> Add `pluralize()` to the catalog kind header component.

Reviewer: We can be a bit less specific about the method used :) Just hoping that pluralisation is a word that our spellchecker knows 🙏 😅 
```suggestion
Fixed Entity kind pluralisation in the `CatalogKindHeader` component.
```
